docs/hugo/content/guide/breaking-changes/breaking-changes-v2.1.0.md

```diff
@@ -30,3 +30,17 @@ allowing references to secrets only if the secret is in the same namespace as th
 This was a security issue which we had to close.
 
 See [#2919](https://github.com/Azure/azure-service-operator/pull/2919)
+
+## ManagedClusters ManagedClusterServicePrincipalProfile.Secret field is now marked as a SecretReference
+
+We have marked `ManagedClusterServicePrincipalProfile.Secret` property to be a secret reference. If `ContainerService/ManagedClusters` resource is installed in your cluster, 
+below steps should be performed before upgrading. If you have `ManagedClusterServicePrincipalProfile.Secret` property set on your `ManagedCluster` resource, follow the steps below:
+
+1. Annotate the resource with `serviceoperator.azure.com/reconcile-policy: skip` to prevent ASO from trying to reconcile the resource while you are upgrading.
+2. Download the current YAML for the resource using `kubectl` if you don't have it elsewhere.
+4. Create a kubernetes secret containing the value for `ManagedClusterServicePrincipalProfile.Secret`.
+5. Edit downloaded YAML in step 3, and add a secret key and name reference. Example [here](https://github.com/Azure/azure-service-operator/blob/main/v2/samples/compute/v1api/v1api20201201_virtualmachine.yaml#L18).
+6. Delete the resource from your cluster using `kubectl delete`. Your Azure resource will be left untouched because of the `reconcile-policy` annotation you added above.
+7. [Upgrade ASO](../../upgrading) in your cluster.
```

Comment:
Use the Hugo function **relref** to construct the URL here.

```suggestion
7. [Upgrade ASO]( {{< relref "upgrading" >}} ) in your cluster.
```
As long as the page name (without the `.md` extension) is unique, Hugo will find it and construct the right relative URL.